### ISSUE TITLE
[Question] Difference between make_vec_env and SubprocVecEnv for multiprocessing

### ISSUE THREAD

**Issue by A-Artemis** [DLR-RM/stable-baselines3#1654]
### ❓ Question

Hi, I am trying to use multiprocessing on my custom gym environment and I am unsure about the parallel learning and the logging.

I first started using:
```python
envs = make_vec_env(env_id=make_env(), n_envs=8, vec_env_cls=SubprocVecEnv)
```
and this seems to work. I see 8 environments spawn and the iterations become an ~8x improvement, I suspect it is doing parallel learning.

But then I came across this [documentation](https://colab.research.google.com/github/Stable-Baselines-Team/rl-colab-notebooks/blob/sb3/multiprocessing_rl.ipynb#scrollTo=TgjfyOTPVxG6)  which shows that I can use:
```python
envs = SubprocVecEnv([make_env() for _ in range(8)])
```

When using option one, my output is showing the rollout values:
```
---------------------------------
| rollout/           |          |
|    ep_len_mean     | 84.9     |
|    ep_rew_mean     | 29.7     |
| time/              |          |
|    fps             | 304      |
|    iterations      | 1        |
|    time_elapsed    | 26       |
|    total_timesteps | 8192     |
---------------------------------
-----------------------------------------
| rollout/                |             |
|    ep_len_mean          | 84.9        |
|    ep_rew_mean          | 29.7        |
| time/                   |             |
|    fps                  | 305         |
|    iterations           | 2           |
|    time_elapsed         | 53          |
|    total_timesteps      | 16384       |
| train/                  |             |
|    approx_kl            | 0.021776147 |
|    clip_fraction        | 0.41        |
|    clip_range           | 0.2         |
|    entropy_loss         | -24.1       |
|    explained_variance   | -0.0814     |
|    learning_rate        | 0.005       |
|    loss                 | 0.347       |
|    n_updates            | 4           |
|    policy_gradient_loss | 0.0183      |
|    std                  | 1           |
|    value_loss           | 2.04        |
-----------------------------------------
```

but when I use the second option, my output does not include any rollout values:
```
-----------------------------
| time/              |      |
|    fps             | 311  |
|    iterations      | 1    |
|    time_elapsed    | 26   |
|    total_timesteps | 8192 |
-----------------------------
-----------------------------------------
| time/                   |             |
|    fps                  | 312         |
|    iterations           | 2           |
|    time_elapsed         | 52          |
|    total_timesteps      | 16384       |
| train/                  |             |
|    approx_kl            | 0.033831578 |
|    clip_fraction        | 0.5         |
|    clip_range           | 0.2         |
|    entropy_loss         | -24.2       |
|    explained_variance   | -0.0934     |
|    learning_rate        | 0.005       |
|    loss                 | 0.273       |
|    n_updates            | 4           |
|    policy_gradient_loss | 0.0275      |
|    std                  | 1.01        |
|    value_loss           | 2.2         |
-----------------------------------------
```

So I have two questions:
- Is `make_vec_env(..., vec_env_cls=SubprocVecEnv)` making use of parallel learning (broadcasting/recieveing) actions and policies? Or is `make_vec_env` just 8 separate environments which are learning individually? (I am assuming that SubprocVecEnv is correctly learning in with other threads [docs](https://stable-baselines3.readthedocs.io/en/master/guide/vec_envs.html))
- And why is there a discrepancy between the logging. Is there an additional step that I am missing?

Here is my `main.py`, 

```python
from typing import Callable
from gymnasium import Env
from osim.osim import OsimModel
from stable_baselines3 import PPO
from stable_baselines3.common.env_util import make_vec_env, SubprocVecEnv
from stable_baselines3.common.logger import configure


def make_env() -> Callable:
    def _init() -> Env:
        env = OsimModel(
            model_path="osim/models/pros.osim",
            data_path="data/83_35/83_35.csv",
            integrator_accuracy=0.01,
            visualize=True,
        )
        env.reset()
        return env

    return _init


if __name__ == '__main__':
    envs = make_vec_env(env_id=make_env(), n_envs=8, vec_env_cls=SubprocVecEnv) # does log rollouts
    # envs = SubprocVecEnv([make_env() for _ in range(8)]) # does not log rollouts
    model = PPO(
        policy="MlpPolicy",
        env=envs,
        learning_rate=0.005,
        n_steps=1024,
        batch_size=512,
        n_epochs=4,
        ent_coef=0.001,
        verbose=True,
    )
    log = configure(folder="./models", format_strings=["stdout", "csv"])
    model.set_logger(logger=log)
    model.learn(total_timesteps=10_000_000, progress_bar=True, log_interval=1)
```

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by qgallouedec** [comment]
> * Is `make_vec_env(..., vec_env_cls=SubprocVecEnv)` making use of parallel learning (broadcasting/recieveing) actions and policies? Or is `make_vec_env` just 8 separate environments which are learning individually? (I am assuming that SubprocVecEnv is correctly learning in with other threads [docs](https://stable-baselines3.readthedocs.io/en/master/guide/vec_envs.html))

1 agent collects interaction with N (=8 here) parallel environments: data collection is parallelized, training the agent is centralized.


> * And why is there a discrepancy between the logging. Is there an additional step that I am missing?

The difference is that with `make_vec_env` (recommended option), the environment is wrapped in a with a `Monitor` wrapper. This wrapper monitors several values, in particular the average reward per episode. This monitored data is then retrieved by the logger to be logged. That why you can see `rollout/ep_len_mean` when you use `make_vec_env`.

**Comment by A-Artemis** [comment]
Thanks for the quick reply and the info!
